src/Parser.hs

```diff
@@ -5,113 +5,112 @@ module Parser
+    parseAllOf :: Char -> Parser Accidental
+    parseAllOf acc = do let constructor = case acc of
+                                            '#' -> AccSharp
+                                            'b' -> AccFlat
+                        accs <- many1 $ char acc
+                        return $ constructor $ length accs
 
 parserRoot :: Parser Root
-parserRoot =
-  do
-    noteChar <- oneOf "ABCDEFG"
-    let note = read [noteChar]
-    Root note <$> parserAccidental
+parserRoot = do noteChar <- oneOf "ABCDEFG"
+                let note = read [noteChar]
+                Root note <$> parserAccidental
 
 parserQuality :: Parser Quality
-parserQuality =
-       parser "^" QMajor
-  <||> parser "M" QMajor
-  <||> parser "-" QMinor
-  <||> parser "m" QMinor
-  <||> parser "dim" QDiminished
-  <||> parser "o" QDiminished
-  <||> parser "aug" QAugmented
-  <||> parser "+" QAugmented
+parserQuality = choice $ parseQualfromString <$> qualStrings
   where
-    parser :: String -> Quality -> Parser Quality
-    parser literal quality =
-      do
-        _ <- string literal
-        return quality
+    qualStrings = [ ("^",   QMajor)
+                  , ("M",   QMajor)
+                  , ("-",   QMinor)
+                  , ("m",   QMinor)
+                  , ("dim", QDiminished)
+                  , ("o",   QDiminished)
+                  , ("aug", QAugmented)
+                  , ("+",   QAugmented )
+                  ]
+
+    parseQualfromString :: (String, Quality) -> Parser Quality
+    parseQualfromString (qualName, quality) = string qualName >> return quality
 
 parserHighestNatural :: Parser HighestNatural
 parserHighestNatural =
   do
-    constructor <- parserMajor
+    constructor <- parseMajToConstructor
     digits <- many1 digit
-    return $ constructor (read digits)
+    return $ constructor $ read digits
   where
-    parserMajor :: Parser (Int -> HighestNatural)
-    parserMajor =
-      do
-        major <- optionMaybe $ string "Maj" <||> string "M" <||> string "^"
-        return $
-          case major of
-            Just _ -> HighestNatural Major
-            Nothing -> HighestNatural NonMajor
+    parseMajToConstructor :: Parser (Int -> HighestNatural)
+    parseMajToConstructor =
+      do major <- optionMaybe $ string "Maj" <||> string "M" <|> string "^"
+         return $ HighestNatural $
+           case major of
+             Just _  -> Major
+             Nothing -> NonMajor
 
 parserSus :: Parser Sus
 parserSus =
-  do
-    msus <- optionMaybe parserSusPresent
-    let sus = maybe NoSus Sus msus
-    return sus
+  do msus <- optionMaybe parserSusPresent
+     return $ maybe NoSus Sus msus
   where
     parserSusPresent :: Parser Int
     parserSusPresent =
-      do
-        _ <- string "sus"
-        number <- optionMaybe $ many1 digit
-        return $ maybe 2 read number
+      do string "sus"
+         number <- optionMaybe $ many1 digit
+         return $ maybe 2 read number
```

Comment:
The only important change here is from `_ <- string "sus"` to simply `string "sus"`; this is essentially _exactly_ the refactor discussed in the comment immediately above this one.